### PR TITLE
Enable program filtering flag in some browser tests and fix screenshots (part 5)

### PR DESCRIPTION
**Pull request by melanie-exygy**
### Description

Enabling the program filtering flag for some test files and updating screenshots so the browser tests pass. This is in preparation for turning on the filtering flag by default in the staging environment.
Also, replacing tests where the flag is disabled with duplicate tests that have the flag enabled.  This is as a clean-up so that we can remove the flag shortly.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [ ] Created unit and/or browser tests which fail without the change (if possible)
- [ ] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [ ] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`
- [ ] Noted in the PR description which, if any, code in this PR was generated by AI.

### CODE REVIEW

File: browser-test/src/applicant/applicant_application_index.test.ts [hunk 1]
```diff
@@ -175,52 +176,6 @@ test.describe('applicant program index page', () => {
     )
   })
 
-  test('categorizes programs for draft and applied applications', async ({
```
Comment (melanie-exygy) [inline browser-test/src/applicant/applicant_application_index.test.ts:178]: **note:** I'm replacing this with the essentially duplicate test below, the only difference being that program filtering is enabled in the other test.  This means that we won't be testing the disabled state anymore, but governments have been running with the flag enabled for months, so that state doesn't exist anymore in practice.  I'll be enabling the flag in all environments by default in the next two weeks and then removing the flag.

**Review by tallulahkay — APPROVED** [review]
